### DISCUSSION TITLE
How do I make regex to ignore whitespace within google searches?

### DISCUSSION THREAD

**Discussion by Hark64** [proginosko/LeechBlockNG#442] [Q&A] [answered]
[This](https://www.tapatalk.com/groups/leechblock/viewtopic.php?p=927#p927) tapatalk post provides a regex expression, ^https?:\/+((www\.)?google\.com)\/search(.*)q=(\w+\++)*(blue|green|red), that allows leechblock to block certain google searches. But it does not handle spaces inside of terms, which show up as a '+' in the url (e.g. blue or blue+red will match but not b+lue, bl+ue, blu+e). 

So I can pretty easily invalidate this regex. How do I get it to ignore spaces in a google search query?

**Comment by proginosko** [comment]
I'm curious. Why would you need to match spaces _within_ search terms?

**Reply by Hark64** [reply]
I think I explained it but if I don't want to allow searches for the word 'blue' I can get around this by just searching for 'b lue' since google will treat it as 'blue' and give me the same results. But the actual url will be ...q=b+lue so the regex will be pretty easily invalidated.

This can of course happen with different misspellings too but generally spelling a word differently will give different results. Whitespace inside search terms will generally still be understood by google however.

**Comment by proginosko** [comment, marked as answer]
Thanks for the explanation. I understand now. I didn't know Google did that.

The solution is messy, but it would look like this:

`^https?:\/+((www\.)?google\.com)\/search(.*)q=(\w+\++)*(b\+*l\+*u\+*e|g\+*r\+*e\+*e\+*n|r\+*e\+*d)`

In sum, put `\+*` where any space might be between letters.

**Reply by Hark64** [reply]
I was hoping that wouldn't be the case but thank you so much, I was bashing my head for hours trying to figure out something. 
I'm glad that there is some way at least to alter the regex.

**Reply by proginosko** [reply]
You're welcome. In principle, there's a regex for everything, just not necessarily a pretty one!
